src/Type/Php/ParseUrlFunctionDynamicReturnTypeExtension.php

```diff
@@ -105,19 +123,43 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 			return $this->componentTypesPairedConstantsForLowercaseString[$componentType->getValue()] ?? new ConstantBooleanType(false);
 		}
 
+		if ($urlType->isUppercaseString()->yes()) {
```

Comment:
Dunno if you shouldn't handle the case 
```
$urlType->isLowercaseString()->yes() && $urlType->isUppercaseString()->yes()
```
before ?

Reply:
Removed in 356e0f810